include/openenclave/internal/str.h

```diff
@@ -294,7 +294,7 @@ MEM_INLINE int str_printf(str_t* str, const char* format, ...)
     }
 
     /* Set the size */
-    str->__mem.__size = r + 1;
+    str->__mem.__size = (size_t)r + 1;
```

Comment:
This is okay in our code but relies on the implicit assumption that we only build on x64 so `size_t` and the size of pointers are the same. Can you please add a static assert for `OE_STATIC_ASSERT(sizeof(size_t) == sizeof(char*));` at the beginning of the file?